core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -297,6 +358,20 @@ public Expression visitAllFields(AllFields node, AnalysisContext context) {
   @Override
   public Expression visitQualifiedName(QualifiedName node, AnalysisContext context) {
     QualifierAnalyzer qualifierAnalyzer = new QualifierAnalyzer(context);
+
+    // check for reserved words in the identifier
+    TypeEnvironment typeEnv = context.peek();
```

Comment:
1. Why TypeEnv differenate symbol Table and resvered symbol table?
2. The logic is similr to visitIdentifier, is it possible to reuse visitIdentifier?

Reply:
Yes.  By splitting them, we can treat them differently (or not) in later steps.  For example, reserved words are only output when explicitly requested - so not in a wildcard projection.

Reply:
1. why not include _id when expand `*`. any concern?

Reply:
Track the discussion on https://github.com/opensearch-project/sql/issues/1506.